src/ui/libs/DatalensChartkit/modules/html-generator/utils.ts

```diff
@@ -11,3 +15,122 @@ export function validateUrl(url: string, errorMsg?: string) {
+    });
+
+    const hasMissedVariables = Object.values(missedVariables).some(
+        (variables) => variables?.length,
+    );
+
+    if (hasMissedVariables) {
+        let msg = '\nInconsistent defining of variables among themes.\n\n';
+        Object.entries(missedVariables).forEach(([themeKey, variables]) => {
+            msg += `Missing variables in ${themeKey} theme:\n${variables.join('\n')}\n\n`;
+        });
+        msg += 'You should define missing variables due to have a consistent theme configuration.';
+        throw new ChartKitCustomError(undefined, {message: msg, details: msg});
+    }
+}
+
+function validateTheme(value: unknown): value is ChartKitHtmlItemTheme {
+    if (!isObject(value)) {
+        const msg = '\n"theme" property should be an object';
+        throw new ChartKitCustomError(undefined, {message: msg, details: msg});
+    }
+    if (!('dark' in value) || !('light' in value)) {
+        const msg = '\nYou should specify "theme.dark" and "theme.light" properties';
+        throw new ChartKitCustomError(undefined, {message: msg, details: msg});
+    }
+    if (
+        !isObject(value.dark) ||
+        !isObject(value.light) ||
+        ('dark-hc' in value && !isObject(value['dark-hc'])) ||
+        ('light-hc' in value && !isObject(value['light-hc']))
+    ) {
+        const msg = '\nTheme properties should be objects';
+        throw new ChartKitCustomError(undefined, {message: msg, details: msg});
+    }
+
+    Object.values(value as ChartKitHtmlItemTheme).forEach((theme) => {
+        validateThemeProperties(theme);
+    });
+
+    validateThemesConsistency(value as ChartKitHtmlItemTheme, ['dark', 'light']);
+
+    return true;
+}
+
+function getThemeSelector(theme: string, dataThemeId: string) {
+    const isHcTheme = theme.endsWith('hc');
+    // Add ".g-root" class to increase specificity for high contrast
+    // theme due to fallback selectors for regular light and dark themes
+    const baseSelector = `${isHcTheme ? '.g-root' : ''}.g-root_theme_${theme} [${ATTR_DATA_CE_THEME}="${dataThemeId}"]`;
+    const baseHCSelector = isHcTheme
+        ? ''
+        : // Add fallback for high contrast theme in order
+          // not to force users to define all available themes
+          `.g-root_theme_${theme}-hc [${ATTR_DATA_CE_THEME}="${dataThemeId}"]`;
+    return baseHCSelector
+        ? `${baseSelector},${baseSelector} *,${baseHCSelector},${baseHCSelector} * {`
+        : `${baseSelector},${baseSelector} * {`;
+}
+
+export function getThemeStyle(value: unknown, dataThemeId: string) {
+    if (!validateTheme(value)) {
+        // Add return for value type narrowing
+        return '';
+    }
+
+    const themes = Object.entries(value).map(([theme, variables]) => {
+        const variableEntries = Object.entries(variables);
+        const style = variableEntries.reduce(
+            (acc, [variable, variableValue], i) => {
+                let nextAcc = `${acc}${variable}:${escape(variableValue)};`;
```

Comment:
Everything should be escaped, including the key. Now it is possible to insert something like

`'--ce-theme</style><style>.smth {display: none;}</style>': '*',`